crates/sequencer/src/runner.rs

```diff
@@ -591,6 +625,10 @@ where
                 txs_to_remove.extend(l1_fee_failed_txs);
 
                 self.mempool.remove_transactions(txs_to_remove.clone());
+                self.telemetry
+                    .targets
+                    .mempool_txs
+                    .dec_by(txs_to_remove.len() as i64);
```

Comment:
I would suggest using pools len function to check the size of the pool, it might be discarding some txs on it's own as well after a while
You'll need to add that function to CitreaMemppol first as it does not exist there